src/main/java/org/mskcc/cbio/portal/scripts/GeneticAlterationIncrementalImporter.java

```diff
@@ -0,0 +1,105 @@
+package org.mskcc.cbio.portal.scripts;
+
+import org.mskcc.cbio.portal.dao.DaoException;
+import org.mskcc.cbio.portal.dao.DaoGeneticAlteration;
+import org.mskcc.cbio.portal.dao.DaoGeneticProfileSamples;
+import org.mskcc.cbio.portal.model.CanonicalGene;
+import org.mskcc.cbio.portal.util.ArrayUtil;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+
+public class GeneticAlterationIncrementalImporter extends GeneticAlterationImporter {
+
+    private final List<Integer> fileOrderedSampleList;
+    private final DaoGeneticAlteration daoGeneticAlteration = DaoGeneticAlteration.getInstance();
+    private final HashMap<Integer, HashMap<Integer, String>> geneticAlterationMap;
+
+    public GeneticAlterationIncrementalImporter(
+            int geneticProfileId,
+            List<Integer> fileOrderedSampleList
+    ) throws DaoException {
+
+        this.geneticProfileId = geneticProfileId;
+        this.geneticAlterationMap = daoGeneticAlteration.getGeneticAlterationMapForEntityIds(geneticProfileId, null);
+        this.fileOrderedSampleList = fileOrderedSampleList;
+
+        ArrayList <Integer> savedOrderedSampleList = DaoGeneticProfileSamples.getOrderedSampleList(this.geneticProfileId);
+        int initialOrderSampleListSize = savedOrderedSampleList.size();
+        geneticAlterationMap.forEach((geneticEntityId, sampleToValue) -> {
+            if (sampleToValue.size() != initialOrderSampleListSize) {
+                throw new IllegalStateException("Number of samples ("
+                        + sampleToValue.size() + ") for genetic entity with id "
+                        + geneticEntityId + " does not match with the number in the inital sample list ("
+                        + initialOrderSampleListSize + ").");
+            }
+        });
+        // add all new sample ids at the end
+        this.orderedSampleList = new ArrayList<>(savedOrderedSampleList);
+        List<Integer> newSampleIds = this.fileOrderedSampleList.stream().filter(sampleId -> !savedOrderedSampleList.contains(sampleId)).toList();
```

Comment:
I think this is order(N^2) time complexity. For lists containing only a thousand or two samples, this may not be a factor, but at some point (millions? tens of millions?) it might become slow. A faster approach would be to first sort each list and then scan each list in parallel to find the diff.
I suppose I am assuming that contains() is an order(N) operation. If the list is also represented with a hashmap for faster contains() execution then perhaps this comment is unsound.

Reply:
Here it is https://github.com/cBioPortal/cbioportal-core/pull/48/commits/9a37bbecf841e15503fddedb55bb48605c552378

Reply:
Ok, that solution is even better. The set lookup can be (in theory) order 1 which give better overall complexity. Thanks.
👍